src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/AuthenticationController.cs

```diff
@@ -340,8 +346,55 @@ private async Task<ActionResult> AuthenticateMaskinportenToken(string originalTo
+                    string enterpriseUserHeader = Request.Headers["X-Altinn-EnterpriseUser-Authentication"];
+                    byte[] decodedCredentials = Convert.FromBase64String(enterpriseUserHeader);
+                    string decodedString = Encoding.UTF8.GetString(decodedCredentials);
+                    string[] decodedStringArray = decodedString.Split(":");
+                    string usernameFromRequest = decodedStringArray[0];
+                    string password = decodedStringArray[1];
+
+                    string bridgeApiEndpoint = _generalSettings.BridgeAuthnApiEndpoint + "enterpriseuser";
+                    EnterpriseUserAuthenticationService enterpriseUserAuthenticationService = new EnterpriseUserAuthenticationService();
+
+                    EnterpriseUserCredentials credentials = new EnterpriseUserCredentials { UserName = usernameFromRequest, Password = password, OrganizationNumber = orgNumber };
+                    ConfiguredTaskAwaitable<HttpResponseMessage> response = enterpriseUserAuthenticationService.GetResponseMessage(credentials, bridgeApiEndpoint);
+
+                    var responseInfo = response.GetAwaiter().GetResult();
+                    string content = await response.GetAwaiter().GetResult().Content.ReadAsStringAsync();
+
+                    if (responseInfo.StatusCode.ToString() == "OK")
+                    {
+                        authenticatemethod = "virksomhetsbruker";
+
+                        UserAuthenticationResult userAuthenticationResult = new UserAuthenticationResult();
+                        userAuthenticationResult = JsonSerializer.Deserialize<UserAuthenticationResult>(content);
+
+                        string userID = userAuthenticationResult.UserID.ToString();
+                        string username = userAuthenticationResult.Username;
+                        string partyId = userAuthenticationResult.PartyID.ToString();
+
+                        claims.Add(new Claim(AltinnCoreClaimTypes.UserId, userID, ClaimValueTypes.Integer32, issuer));
+                        claims.Add(new Claim(AltinnCoreClaimTypes.UserName, username, ClaimValueTypes.String, issuer));
+                        claims.Add(new Claim(AltinnCoreClaimTypes.PartyID, partyId, ClaimValueTypes.Integer32, issuer));
+                    }
+                    else if (responseInfo.StatusCode.ToString() == "BadRequest")
+                    {
+                        ObjectResult result = StatusCode(400, responseInfo.ReasonPhrase);
+                        return result;
+                    }
+                    else if (responseInfo.StatusCode.ToString() == "TooManyRequests")
+                    {
+                        ObjectResult result = this.StatusCode(429, string.Format("User Locked out. Try again in {0} hours, {1} minutes and {2} seconds.", responseInfo.Headers.RetryAfter.Delta.Value.Hours, responseInfo.Headers.RetryAfter.Delta.Value.Minutes, responseInfo.Headers.RetryAfter.Delta.Value.Seconds));
```

Comment:
Instead of overriding ReasonPhrase (which is a dubious practice from Altinn2) that client implementations tend to ignore (they just look at the status code), just use `this.StatusCode(429)` and add the Retry-After header as received from bridge as-is (add a null check). This will avoid forcing clients to implement a custom reason-phrase parser to figure out when it's ok to try again, and we avoid breaking this API if bridge at some points stops sending Retry-After.

Reply:
Rewritten responsehandling in latest commit.